nav2_map_server/CMakeLists.txt

```diff
@@ -14,15 +14,27 @@ find_package(std_msgs REQUIRED)
 find_package(tf2 REQUIRED)
 find_package(nav2_util REQUIRED)
 find_package(GRAPHICSMAGICKCPP REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(pcl_conversions REQUIRED)
```

Comment:
If you're adding a dependency on PCL, I would recommend making the 3d map server a separate package - this will make it easier to skip building that package (and the PCL dependency). We spent quite a bit of time to avoid having a dependency on PCL in the ROS1 navigation stack -- because these packages were consistently a blocker for releasing into new rosdistros.

Reply:
I think we only need the IO and common modules, and we could have those added as separate rosdep keys so we don't bring in all of PCL (which I agree, bad). Keep in mind pcl_conversions (https://github.com/ros-perception/perception_pcl/blob/melodic-devel/pcl_conversions/package.xml#L22) already brings in all of PCL, so its not like adding pcl here outright is that much worse.

I agree we should consider if this should be a new package, especially given the state of the PR, it may be beneficial to have a clean slate. Long term I would have liked 1 map server to do both as we're adding the semantic map server items, but there may be enough differences requiring separation.